File: CSC330/cnotsosharp/cSharp/collapseFunction.cs
using System;

//This is the collapse function for the Group Project in CSC330
//This program will take in a number and perform the collapse of it and return the result

namespace collapseFunction
{
  public class collapseFunction //Indicates the beginning of our collapse function
  {
    public static void Main(String[] args) //Main runner of the class
    {
      Console.WriteLine("Please input the number that you would like to find the collapse of: "); //Asks the user for their input
      int number = int.Parse(Console.ReadLine()); //Takes the user's number and stores it to use
      Console.WriteLine("The total collapse of " + number + " is " + collapse(number)); //Finds the collapse of our function and prints it
    }
    
    public static int collapse(int checking) //This recursive function will take in an integer value and return the collapsed form of it
    {
      if(checking < 10) //If we have a single digit number being checked
      {
        return checking; //Returns our final result
      }
      else //If the number still has multiple digits
      {
        int sum = 0; //Creates a variable to hold the sum of all the single digits in our number
        while(checking > 9) //While checing is still multidigit
        {
          sum += checking % 10; //Adds the last digit of checking to the sum
          checking = checking / 10; //Cuts off the last digit of checking 
        }
        sum += checking; //Adds the final digit of the checking number to the sum variable
        return collapse(sum); //Recursively runs collapse with the new number to check 
      }
    }
  }
}

File: CSC330/cnotsosharp/rust/collapseFunction.rs


//This is the Rust version of the collapse function for the CSC 330 Group Project

fn main() //This is the main function we will use to run the collapse function
{
  let mut number_string = String::new(); //Creates a string we will use to hold the user's input
  println!("Please enter the number you will be collapsing: "); //The terminal asks the user for the number they want collapsed
  std::io::stdin().read_line(&mut number_string); //This reads in the user's input and stores their number;
  let number:i32 = number_string.trim().parse().expect("Input not an integer"); //This integer will hold the number we are collapsing
  println!("{}", collapse(number).to_string()); //Finds and prints the collapsed version of the user given number
}

fn collapse(mut num:i32) -> i32 //This function takes in an integer and performs the collapse function on it before returning it
{
  if num < 10 //If the number we are checking is a single digit
  {
    return num; //Returns the number
  }
  else
  {
    let mut sum:i32 = 0; //Creates a mutable integer to hold the sum of all the single digits in a multidigit number
    while num > 9 //While the checked number has more than one digit
    {
      sum = sum + (num % 10); //Adds the last digit of num to the sum
      num = num / 10; //Cuts off the last digit of num for addition
    }
    sum = sum + num; //Adds num to the sum
    return collapse(sum); //Runs the collapse of and returns the collapse of the sum
  }
}